### PR TITLE
Use ExtensiblePayload in consensus

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Review by vncoelho — COMMENTED** [review]
I need to understand this concept of `ExtensiblePayload`.
I can try to fix, but may need some couple of days @erikzhang,3-5.

If someone can fix before it I can double check.

**Review by vncoelho — COMMENTED** [review]
As far as I could investigate, @erikzhang, there is a problem with the `RecoveryMessage`.

Basically, I commented all tests...aehauheaue
But kept `ConsensusService_SingleNodeActors_OnStart_PrepReq_PrepResponses_Commits`, which checks a lot of things about the consensus flow.

There is something going on that CommitPayload is not being recovered.

**Comment by vncoelho** [comment]
I am sending the commit here and will keep uncommenting  as soon as we advance.
For now, the debug should focus on making the commit payloads to be `3` after the recovery.

**Comment by vncoelho** [comment]
```cs
                    Console.WriteLine($"\nHI INSIDE RECOVERY IF ...{commitPayloads.Length}");
                    totalCommits = commitPayloads.Length;
                    foreach (ExtensiblePayload commitPayload in commitPayloads)
                        if (ReverifyAndProcessPayload(commitPayload)) validCommits++;
                    Console.WriteLine($"\nVALID COMMITS ...{validCommits}");    
                    
                     HI INSIDE RECOVERY IF ...4
 
                    VALID COMMITS ...0

```                    
Something is happening on `ReverifyAndProcessPayload`

File: src/neo/Network/P2P/MessageCommand.cs [hunk 1]
```diff
@@ -38,8 +38,6 @@ public enum MessageCommand : byte
         Transaction = 0x2b,
         [ReflectionCache(typeof(Block))]
         Block = 0x2c,
-        [ReflectionCache(typeof(ConsensusPayload))]
-        Consensus = 0x2d,
         [ReflectionCache(typeof(ExtensiblePayload))]
         Extensible = 0x2e,
```
Comment (shargon) [inline src/neo/Network/P2P/MessageCommand.cs:42]:
```suggestion
        Extensible = 0x2d,
```
?

Comment (erikzhang) [reply src/neo/Network/P2P/MessageCommand.cs:42]: `0x2e` is okay.

**Comment by erikzhang** [comment]
https://github.com/neo-project/neo/blob/83857d232e755cbc3848a9158e96b54cdfcedb09/src/neo/Network/P2P/Payloads/ExtensiblePayload.cs#L95

The problem is that the mocked validators are not in the whitelist.

**Comment by shargon** [comment]
> The problem is that the mocked validators are not in the whitelist.

I will add them with reflection

**Comment by shargon** [comment]
UT Fixed, we should create the module before merge it in order to test it well

**Comment by erikzhang** [comment]
We can test it without module.

**Review by vncoelho — COMMENTED** [review]
Such a beautiful work, great job @shargon.

**Comment by erikzhang** [comment]
@neo-project/ngd-shanghai Test?

**Comment by erikzhang** [comment]
I will start creating the module after merging this.

**Review by shargon — APPROVED** [review]
It's good for me, we can test it later

**Pull request merged**